tests/rados/test_mgr_daemon.py

```diff
@@ -0,0 +1,134 @@
+"""
+Module to test MGR functionalities
+
+"""
+
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.mgr_workflows import MgrWorkflows
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH manager daemon functional testing
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_nodes = ceph_cluster.get_nodes(role="mgr")
+    mgr_obj = MgrWorkflows(node=cephadm)
+
+    log.info("Scenario 1: Verification of MGR fail scenario")
+    # Get the active MGR name
+    mgr_stats = mgr_obj.get_mgr_stats()
+    active_mgr = mgr_stats["active_name"]
+    # Make active mgr fail
+    mgr_obj.set_mgr_fail(active_mgr)
+    # Check the new mgr is activated
+    mgr_stats = mgr_obj.get_mgr_stats()
+    new_active_mgr = mgr_stats["active_name"]
+
+    if new_active_mgr is active_mgr:
+        log.error("After the MGR failure cluster ")
+        return 1
+    log.info(f"The new active manager is -{new_active_mgr}")
+    # Check the ceph status
+    rados_obj.log_cluster_health()
+    log.info("End of Scenario-1: Verification of MGR fail scenario passed")
+
+    log.info("Scenario 2: Verification of MGR restart scenario")
+    # Restart MGR and check
+    cephadm.shell(["ceph orch restart mgr"])
```

Comment:
How are we confirming that the Mgr daemons have actually restarted? 

in the past, we have seen that exact same issue, where OSD daemons were not restarted, post issuing ` orch restart osd ` commands.

it would be best if we collect the stats from ` ceph orch ps ` before and after the restart command is executed, and verify if the service was restarted.

Reply:
I removed the command and used the existing method -  restart_daemon_services